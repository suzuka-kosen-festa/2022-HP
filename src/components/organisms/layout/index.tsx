import type { FC } from "react";
import tw, { css } from "twin.macro";
import { useRoute } from "wouter";
import { useInView } from "react-intersection-observer";
import type { LayoutProperties } from "./types/model";
import { HeroSection } from "../hero-section";
import { Footer } from "../footer";
import { Header } from "../header";

const LayoutBox = tw.div`min-h-screen bg-white`;

const MainBox = tw.div`flex flex-col items-center relative z-10 space-y-6 sm:space-y-16`;

const MainWrapper = tw.div`mx-auto w-full max-w-main md:max-w-main-md  space-y-10 sm:space-y-20`;

const lists = [
  {
    link: "/",
    color: tw`text-primary-yellow`,
    name: "Top",
  },
  {
    link: "/member",
    color: tw`text-primary-red`,
    name: "Members",
  },
  {
    link: "/timetable",
    color: tw`text-primary-green`,
    name: "Schedule",
  },
  {
    link: "/bazaar",
    color: tw`text-primary-purple`,
    name: "Bazaar",
  },
  {
    link: "/sponsors",
    color: tw`text-primary-blue`,
    name: "Sponsors",
  },
];

const Layout: FC<LayoutProperties> = ({ children }) => {
  const [match] = useRoute("/");
  const { ref, inView } = useInView({
    rootMargin: "-50% 0px",
  });
  return (
    <LayoutBox>
      {match ? (
        <Header
          headerLinks={lists}
          css={
            inView
              ? css`
                  animation-name: fadein;
                  animation-duration: 1s;
                  animation-iteration-count: 1;
                  animation-fill-mode: forwards;
                  @keyframes fadein {
                    0% {
                      opacity: 0;
                      display: none;
                    }
                    100% {
                      opacity: 1;
                      display: flex;
                    }
                  }
                `
              : tw`hidden`
          }
        />
      ) : (
        <Header headerLinks={lists} />
      )}
      {match ? <HeroSection date="2022.10.30-31" title="第57回鈴鹿高専祭" /> : null}
      <MainBox css={match ? tw`pt-6 md:pt-16` : tw`pt-main md:pt-main-md`}>
        <MainWrapper ref={ref}>{children}</MainWrapper>
        <Footer link="https://github.com/suzuka-kosen-festa/2022-HP/blob/main/LICENSE">
          第57回鈴鹿高専祭実行委員会
        </Footer>
      </MainBox>
    </LayoutBox>
  );
};

export { Layout };
