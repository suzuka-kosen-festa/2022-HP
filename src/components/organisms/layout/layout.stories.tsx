import type { ComponentStoryObj, ComponentMeta } from "@storybook/react";
import { ComponentPropsWithoutRef } from "react";
import { Layout } from ".";

type T = typeof Layout;
type Story = ComponentStoryObj<T>;
type Meta = ComponentMeta<T>;

const args: ComponentPropsWithoutRef<T> = {
  children: (
    <div>
      <h1>Test</h1>
      <p>this is test</p>
      <p>this is test</p>
      <p>this is test</p>
      <p>this is test</p>
      <p>this is test</p>
      <p>this is test</p>
      <p>this is test</p>
      <p>this is test</p>
      <p>this is test</p>
      <p>this is test</p>
      <p>this is test</p>
      <p>this is test</p>
      <p>this is test</p>
      <p>this is test</p>
      <p>this is test</p>
      <p>this is test</p>
      <p>this is test</p>
      <p>this is test</p>
      <p>this is test</p>
      <p>this is test</p>
      <p>this is test</p>
    </div>
  ),
};

export default {
  args,
  argTypes: {
    children: {
      description: "ReactNode which is inside layout",
      control: {
        type: "object",
      },
    },
  },
  component: Layout,
} as Meta;

export const Default: Story = {};
