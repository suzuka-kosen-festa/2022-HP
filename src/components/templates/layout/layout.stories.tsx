import type { ComponentStoryObj, ComponentMeta } from "@storybook/react";
import type { ComponentPropsWithoutRef } from "react";
import { Layout } from ".";

type T = typeof Layout;
type Story = ComponentStoryObj<T>;
type Meta = ComponentMeta<T>;

const args: ComponentPropsWithoutRef<T> = {
  children: <section />,
};

export default { args, argTypes: { children: { description: "Main content" } }, component: Layout } as Meta;

export const Default: Story = {};
