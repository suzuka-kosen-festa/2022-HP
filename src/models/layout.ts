import type { ComponentProps, ReactHTML, ReactNode } from "react";

export type LayoutProperties = ComponentProps<ReactHTML["div"]> & {
  children: ReactNode;
};
