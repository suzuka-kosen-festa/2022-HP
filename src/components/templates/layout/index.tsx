import React from "react";
import tw from "twin.macro";
import type { LayoutProperties } from "../../../models";
import { Footer } from "../../organisms/footer";
import {HeroSection} from "../../organisms/hero-section";

const LayoutBox = tw.div`min-h-screen bg-background-gray`;

const MainBox = tw.div`flex flex-col`;

const Main = tw.div`mx-[calc(800vw / 63)]`;

const Layout: React.FC<LayoutProperties> = ({ children, ...rest }) => (
  <LayoutBox {...rest}>
    <MainBox>
      <HeroSection date={"2022.10.30-31"} title={"第57回鈴鹿高専祭"} />
      <Main role={"main"}>{children}</Main>
      <Footer link={"https://github.com/suzuka-kosen-festa/2022-HP/blob/main/LICENSE"}>
        &copy; 第57回鈴鹿高専祭実行委員会
      </Footer>
    </MainBox>
  </LayoutBox>
);

export { Layout };
