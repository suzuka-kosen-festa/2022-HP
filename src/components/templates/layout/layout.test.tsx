import { composeStories } from "@storybook/testing-react";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import * as stories from "./layout.stories";

const { Default } = composeStories(stories);

describe("(components) templates/layout", () => {
  test("to be templates", () => {
    const { container } = render(<Default />);
    expect(container).toBeTemplate();
  });
  test("take snap shot", () => {
    const { container } = render(<Default />);
    expect(container).toMatchSnapshot();
  });
});
