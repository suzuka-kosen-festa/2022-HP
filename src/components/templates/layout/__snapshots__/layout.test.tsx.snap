// Vitest Snapshot v1

exports[`(components) templates/layout > take snap shot 1`] = `
<div>
  <div
    class="css-1hq8773"
  >
    <div
      class="css-h1gvs3"
    >
      <header
        class="css-j7qwjs"
      >
        <div
          class="css-bg20hz-HeroSection"
        >
          <p
            class="css-19ggdn4-HeroSection"
          >
            第57回鈴鹿高専祭
          </p>
          <img
            alt="第57回鈴鹿高専祭、Re:ROADのロゴ"
            class="css-1ysdsd"
            height="247"
            src="/statics/logo.webp"
            width="1212"
          />
          <p
            class="css-19ggdn4-HeroSection"
          >
            2022.10.30-31
          </p>
        </div>
      </header>
      <main
        class="css-p8s7gt"
      >
        <section />
      </main>
      <footer
        class="css-1a88kbw"
        role="contentinfo"
      >
        <a
          class="css-154ykdi-Footer"
          href="https://github.com/suzuka-kosen-festa/2022-HP/blob/main/LICENSE"
          rel="noopener noreferrer"
          role="link"
          tabindex="0"
          target="_blank"
        >
          © 
          第57回鈴鹿高専祭実行委員会
        </a>
      </footer>
    </div>
  </div>
</div>
`;
